### ISSUE TITLE
Add support for Dark mode and automatically follow the operating system preference, plus provide a toggle

### ISSUE THREAD

**Issue by TimLethbridge** [umple/Umple#2253]
Many developers prefer dark mode

Umpleonline should follow the mode used in the operating system by using CSS the @media (prefers-color-scheme: dark) rule to check for a user's preference, and the function matchMedia in JavaScript, makes it possible to test against a user’s preferred scheme in scripts.

Note that this may or may not be tricky. The CodeMirror colouring of text may need to be adjusted so the contrast is sufficient and looks good. The icons at the top of the screen my need alternative versions.

Here are some tips

https://www.hanselman.com/blog/how-to-detect-if-the-users-os-prefers-dark-mode-and-change-your-site-with-css-and-js

Why do this? Most code editors allow dark mode and there is a strong preference among many people to use dark mode when editing.

Dealing with generated svg might be an issue. It will likely be necessary to adjust the compiler so that it can generate dark-mode svgs. See the following for starter thoughts. This might be done as an initial sub-issue. This could be a suboption available to all the graphviz generators.

Here is a suggestion to indicate how to do this.

https://cprimozic.net/notes/posts/basic-graphviz-dark-theme-config/

https://stackoverflow.com/questions/67187091/creating-svg-that-appears-black-in-light-mode-and-light-in-dark-mode
